### PR TITLE
Email Subject added in the subscription

### PR DESCRIPTION
**Pull request by saif-ericsson**
### Description of the Change

User can add email subject from the frontend
### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by:

### CODE REVIEW

File: src/main/resources/templates/subscription.html [hunk 1]
```diff
@@ -156,7 +156,22 @@ <h3 class="modal-title text-center" id="formHeader">Subscription Form</h3>
...
                             </div>
-
+                            
+                             <!-- Form post emailSubject field -->                            
+                            <div class="form-group" data-bind="visible: $data.notificationType() == 'MAIL'">
+                                <label class="pl-1 control-label font-weight-bold">Subject</label>
+                                <img class="cursor-pointer popup-large" id="restPostMediaTypeInfo" alt="RestPostMediaType Info" src="assets/images/information.png" data-toggle="tooltip" data-placement="top"
+                                    title=
+                                        "This is your choice for the email notification subject. <br>
+                                        Default value is 'Email Subscription Notification' <br>
```
Comment (a-breid-ericsson) [inline src/main/resources/templates/subscription.html]: You should not say what a default value is, just point that a default value may have been set by the admin, If the default value has been changed by the admin we lie to the users here.

File: src/main/resources/templates/subscription.html [hunk 2]
```diff
@@ -156,7 +156,22 @@ <h3 class="modal-title text-center" id="formHeader">Subscription Form</h3>
                                     </label>
                                 </div>
                             </div>
-
+                            
+                             <!-- Form post emailSubject field -->                            
+                            <div class="form-group" data-bind="visible: $data.notificationType() == 'MAIL'">
```
Comment (a-breid-ericsson) [inline src/main/resources/templates/subscription.html]: if you merge with current master you will see that a H5 class is added for the labels.

File: src/main/resources/templates/subscription.html [hunk 3]
```diff
@@ -293,7 +308,7 @@ <h5>OR</h5>
                                 <img class="cursor-pointer" alt="Requirement Information" src="assets/images/information.png" data-toggle="tooltip" data-placement="top"
                                     title="Info: 'AND' is used between Conditions in Requirement groups, and 'OR' is used between Requirement groups."/>
                             </div>
-                            <div class="p-1 border border-primary form-group">
+                            <div class="p-1 border border-primary form-group">
```
Comment (a-breid-ericsson) [inline src/main/resources/templates/subscription.html]: You should try to enable trim trailing white spaces in your editor, here is some trailing white spaces.

File: src/main/resources/templates/subscription.html [hunk 4]
```diff
@@ -156,7 +156,22 @@ <h3 class="modal-title text-center" id="formHeader">Subscription Form</h3>
...
+                            <div class="form-group" data-bind="visible: $data.notificationType() == 'MAIL'">
+                                <label class="pl-1 control-label font-weight-bold">Subject</label>
+                                <img class="cursor-pointer popup-large" id="restPostMediaTypeInfo" alt="RestPostMediaType Info" src="assets/images/information.png" data-toggle="tooltip" data-placement="top"
+                                    title=
+                                        "This is your choice for the email notification subject. <br>
+                                        Default value is 'Email Subscription Notification' <br>
+                                        which is configured in the property file" />                                
+                                <div >
+                                    <label>
+                                        <input name="emailSubject"  id="emailSubjectID" placeholder="emailSubject" type="text"/>
```
Comment (a-breid-ericsson) [inline src/main/resources/templates/subscription.html]: We should discuss the width of the input in the team. All other inputs stretch 100% ish, Maybe this should be set here to, or Others may be changed.

File: src/main/resources/templates/subscription.html (hunk 1 above)
Comment (saif-ericsson) [reply src/main/resources/templates/subscription.html]: I have changed the text ...

File: src/main/resources/templates/subscription.html (hunk 2 above)
Comment (saif-ericsson) [reply src/main/resources/templates/subscription.html]: done

File: src/main/resources/templates/subscription.html (hunk 4 above)
Comment (saif-ericsson) [reply src/main/resources/templates/subscription.html]: done

File: src/main/resources/templates/subscription.html [hunk 5]
```diff
@@ -159,36 +159,52 @@ <h3 class="modal-title text-center" id="formHeader">Subscription Form</h3>
...
+                                    title=
+                                        "This is your choice for the email notification subject. <br>
+                                        Default value is set by admin in property file" />                                
+                                <div>
+                                    <input name="emailSubject"  id="emailSubjectID" placeholder="emailSubject" class="form-control" type="text"/>                                    
+                                </div>
+                            </div> 
+                            
+                            <div class="p-1 border border-primary form-group">
+<<<<<<< HEAD
```
Comment (J-Binieda-ericsson) [inline src/main/resources/templates/subscription.html]: Resolve merge conflicts and remove those extra lines

File: src/main/resources/static/resources/subscription_templates.js [hunk 6]
```diff
@@ -90,6 +90,7 @@ template_vars["templateEmailTrigger"] =
                     "created": "",
                     "notificationMeta" : "mymail@company.com",
                     "notificationType" : "MAIL",
+                    "emailSubject" : "",
```
Comment (tobiasake) [inline src/main/resources/static/resources/subscription_templates.js]: This needs to added in EI Backend subscription template file as well.

File: src/main/resources/templates/subscription.html [hunk 7]
```diff
@@ -187,6 +200,7 @@ <h3 class="modal-title text-center" id="formHeader">Subscription Form</h3>
                                    <tbody data-bind="foreach: notificationMessageKeyValues">
                                        <tr>
                                            <td valign="top">
+
```
Comment (J-Binieda-ericsson) [inline src/main/resources/templates/subscription.html]: Remove this empty row

File: src/main/resources/static/resources/subscription_templates.js (hunk 6 above)
Comment (saif-ericsson) [reply src/main/resources/static/resources/subscription_templates.js]: If Anders havent already it done, I will do it in a separate commit.

File: src/main/resources/templates/subscription.html [hunk 8]
```diff
@@ -162,6 +162,18 @@ <h3 class="modal-title text-center" id="formHeader">Subscription Form</h3>
...
                                </div>
+                               
+                               <!-- Form post emailSubject field -->                            
+                              <div class="form-group" data-bind="visible: $data.notificationType() == 'MAIL'">
+                                <label class="h5 control-label font-weight-bold">Subject</label>
+                                <img class="cursor-pointer popup-large" id="restPostMediaTypeInfo" alt="RestPostMediaType Info" src="assets/images/information.png" data-toggle="tooltip" data-placement="top"
+                                    title=
+                                        "This is your choice for the email notification subject. <br>
+                                        Default value is set by admin in property file" />                                
+                                <div>
```
[the remark this answers is not in the document]
Comment (saif-ericsson) [reply src/main/resources/templates/subscription.html]: I have changed the text

File: src/main/resources/static/resources/subscription_templates.js (hunk 6 above)
Comment (tobiasake) [reply src/main/resources/static/resources/subscription_templates.js]: ok

**Pull request merged**